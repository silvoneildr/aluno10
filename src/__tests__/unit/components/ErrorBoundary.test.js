import React from 'react';
import ErrorBoundary from 'Components/ErrorBoundary';
import { render } from 'react-testing-library';

const ThrowError = ({ message }) => {
  throw message;
};

describe('ErrorBoundary', () => {
  it('should render without error', () => {
    const { container } = render(
      <ErrorBoundary>
        <p>Hellor World</p>
      </ErrorBoundary>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render error', () => {
    const { container } = render(
      <ErrorBoundary>
        <ThrowError message="Erro" />
      </ErrorBoundary>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
