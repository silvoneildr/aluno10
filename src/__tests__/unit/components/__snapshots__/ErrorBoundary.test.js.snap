// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should render error 1`] = `
<h1>
  Something went wrong.
</h1>
`;

exports[`ErrorBoundary should render without error 1`] = `
<p>
  Hellor World
</p>
`;
